pi_ldapproxy/proxy.py

```diff
@@ -218,19 +226,25 @@ def handleBeforeForwardRequest(self, request, controls, reply):
         :return:
         """
         if isinstance(request, pureldap.LDAPBindRequest):
-            if self.bound:
-                log.warn('Rejected a second bind request in the same connection')
-                self.send_bind_response((False, 'Reusing connections currently unsupported.'), request, reply)
-                return None
-            elif request.dn == '':
+            if self.received_bind_request:
+                # We have already received a bind request in this connection!
+                if self.factory.allow_connection_reuse:
+                    # We need to reset the state before further processing the request
+                    log.info('Reusing LDAP connection, resetting state ...')
+                    self.reset_state()
+                else:
+                    log.warn('Rejected a second bind request in the same connection')
+                    self.send_bind_response((False, 'Reusing connections currently unsupported.'), request, reply)
```

Comment:
The text is wrong. Reusing a connection would be supported. But it is not configured.

"Reusing connection is not configured. Please see config option `allow-connection-reuse`."

Reply:
Oh right, thanks